src/lib/progress-spinner/progress-spinner.ts

```diff
@@ -147,7 +168,10 @@ export class MatProgressSpinner extends _MatProgressSpinnerMixinBase implements
   constructor(public _elementRef: ElementRef,
               platform: Platform,
               @Optional() @Inject(DOCUMENT) private _document: any,
-              @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode?: string) {
+              // @deletion-target 7.0.0 _animationMode and _defaults parameters to be made required.
+              @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode?: string,
+              @Inject(MAT_PROGRESS_SPINNER_DEFAULT_OPTIONS)
+                  private _defaults?: MatProgressSpinnerDefaultOptions) {
```

Comment:
Should we have our own defaults value in a instance of `MatProgressSpinnerDefaultOptions`?

Then we can just `Object.assign` it, rather than holding our defaults inline in location.

In the file above the class:
```ts
export interface MatProgressSpinnerDefaultOptions {
  /** Diameter of the spinner. */
  diameter?: number;
  /** Width of the spinner's stroke. */
  strokeWidth?: number;
}

const BASE_OPTIONS  = {
  diameter: 100,
  strokeWidth: 10
};
```

In the class:
```ts
export class MatProgressSpinner 
  ...
  private defaults: MatProgressSpinnerDefaultOptions
  ...
  constructor(
  ...
  @Inject(MAT_PROGRESS_SPINNER_DEFAULT_OPTIONS) defaults: MatProgressSpinnerDefaultOptions) {
    this.defaults = Object.assign(BASE_OPTIONS, defaults);
  ...
  }
...
```

Reply:
Our defaults are provided as the `MAT_PROGRESS_SPINNER_DEFAULT_OPTIONS` token. I kept the inline default until we can make the `defaults` a required parameter.

Reply:
My concern here is that since the interface for `MatProgressSpinnerDefaultOptions` has all optional properties, that someone who provides a MatProgressSpinnerDefaultOptions, would not provide everything thats needed.